tsl/src/compression/compression.h

```diff
@@ -400,9 +400,8 @@ consumeCompressedData(StringInfo si, int bytes)
 }
 
 /*
- * Normal compression uses 1k rows, but the regression tests use up to 1015.
  * We use this limit for sanity checks in case the compressed data is corrupt.
  */
-#define GLOBAL_MAX_ROWS_PER_COMPRESSION 1015
+#define GLOBAL_MAX_ROWS_PER_COMPRESSION INT16_MAX
```

Comment:
I am wondering if these large batches would be compatible with the TAM implementation indexes. As far as I remember, we have only a limited number of bits to address the tuples.

Reply:
Hard to tell until we have a design doc at least. This refactoring doesn't really enforce a lower limit on batch sizes, so TAM tables can use a low limit if they want to.

Reply:
I am a bit worried about the migration path. If we have larger batches, the compressed table might not be processable using TAM (or we could not build an index). So, it might be good to know more about the requirements. What do you think @mkindahl / @erimatnor?

Reply:
For TAM indexing we need to encode a value's index in a compressed batch into the TID, which has limited number of bits (and we also need block and offset in there). So, yes, there's a limit for sure. However, we could always enforce a lower batch size in TAM if we rewrite the table.

It would be interesting to know the real-world benefits of larger batch sizes.